frontend/src/components/header/Header.jsx

```diff
@@ -1,10 +1,83 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { ReactComponent as Logo } from 'assets/images/logo.svg';
 import { NavLink } from 'react-router-dom';
 import TelegramLogin from '../Telegram/TelegramLogin';
 import './header.css';
 
 function Header({ walletId, onConnectWallet, onLogout, tgUser, setTgUser }) {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  // Close menu when route changes
+  useEffect(() => {
+    setIsMenuOpen(false);
+  }, [window.location.pathname]);
+
+  // Prevent scroll when mobile menu is open
+  useEffect(() => {
+    if (isMenuOpen) {
+      document.body.style.overflow = 'hidden';
+    } else {
+      document.body.style.overflow = 'unset';
+    }
+    return () => {
+      document.body.style.overflow = 'unset';
+    };
+  }, [isMenuOpen]);
+
+  const toggleMenu = () => {
+    setIsMenuOpen(!isMenuOpen);
```

Comment:
could you please create custom hook in `hooks` directory called `useLockBodyScroll` and basically use the exact hook, that was recently created (not merged yet) https://github.com/djeck1432/spotnet/pull/167/files#diff-e3912185f39bc0891d631dd214c08db009ca6edbcbd12a1717fa1afec9b82769
Thx